corehq/apps/groups/dbaccessors.py

```diff
@@ -0,0 +1,45 @@
+import settings
+
+
+def group_by_domain(domain, include_docs=True, **kwargs):
+    from corehq.apps.groups.models import Group
+    return Group.view(
+        'groups/by_domain',
+        key=domain,
+        include_docs=include_docs,
+        **kwargs
+    )
```

Comment:
I've been making an effort to have dbaccessors

1. have an explicit interface (i.e. not use kwargs)
2. not couple anything to the couch API (i.e. not let people pass in reduce=True to completely change the behavior of the function from fetching documents to aggregating)
3. return objects rather than a couchdbkit ViewResult or other couch- or couchdbkit-coupled object type

This has some pretty specific benefits:
1. It means you can write tests for your function that test the specific, limited, and small functionality it claims to provide
2. You can tell just by looking at the function what kind of thing it's going to return without having to examine all of the arguments closely and use you're knowledge of the couchdb API
3. You can deal with the return value of the function directly, instead of having to all `.all()` on it, or parse out [r['id'] for r in result] or any such thing, or be even momentarily confused about whether you have to or not.

If you agree with these ideas, it might be worth looking at all of the places `group_by_domain` is called, seeing exactly what functionality is called for, and limiting the function's functionality to specifically that. If two different kinds of things are needed (counting groups in a domain, fetching groups in a domain), you can make each of those separate functions. Let me know what you think!